package/MDAnalysis/auxiliary/base.py

```diff
@@ -0,0 +1,659 @@
+    n_steps : int
+        Total number of auxiliary steps.
+    n_cols : int
+        Number of columns of data for each auxiliary step.
+    time : float
+        Time of current auxiliary step, as read from data (if present) or 
+        calculated using `dt` and `initial_time`.
+    step_data : ndarray
+        Value(s) from the auxiliary data column(s) of interest (per `data_cols`) 
+        for the current step.
+      
+    ts_data : ndarray
+        List of 'step_data' from each auxiliary step assigned to the last-read
+        trajectory timestep.
+    ts_diff : list
+        List of difference in time between the last-read trajectory timestep
+        and each auxiliary step assigned to it.
+    ts_rep : list of float
+        Represenatative value of auxiliary data for current trajectory timestep.
+
+    Note
+    ----
+    Auxiliary data are assumed to be time ordered and contain no duplicates.
+    It step time is read from the data, it is assumed to be in ps units.
+    """
+
+    # update when add new options
+    represent_options = ['closest', 'average']
+
+    # list of attributes required to recreate the auxiliary
+    required_attrs = ['_represent_ts_as', 'cutoff', '_dt', '_initial_time',
+                      '_time_col', '_data_cols', 'constant_dt', 'auxname', 
+                      'format', '_data_input']
+      
+    def __init__(self, represent_ts_as='closest', auxname=None, cutoff=-1, 
+                 dt=1, initial_time=0, time_col=None, data_cols=None, 
+                 constant_dt=True):
+        # allow auxname to be optional for when using reader separate from
+        # trajectory.
+        self.auxname = auxname
+
+        self.represent_ts_as = represent_ts_as
+
+        self.cutoff = cutoff
+
+        # set initially to avoid error on first read
+        self.n_cols = None
+        self._data_cols = []
+        self._time_col = None
+
+        self.ts_data = None
+        self.ts_rep = None
+
+        self._initial_time = initial_time
+        self._dt = dt
+        self.constant_dt = constant_dt
+
+        self.step = -1
+        self._read_next_step()
+        self.n_cols = len(self._data)
+
+        self.time_col = time_col
+        if data_cols:
+            self.data_cols = data_cols
+        else:
+            self.data_cols = [i for i in range(self.n_cols) 
+                              if i != self.time_col]
+
+        # get dt and initial time from auxiliary data (if time included)
+        if self.time_col is not None and self.constant_dt:
+            self._initial_time = self.time
+            self._read_next_step()
+            self._dt = self.time - self._initial_time
+            self.go_to_first_step()
+
+    def __len__(self):
+        """ Number of steps in auxiliary data. """
+        return self.n_steps
+
+    def next(self):
+        """ Move to next step of auxiliary data. """
+        self._read_next_step()
+        return {'time': self.time, 'data': self.step_data}
```

Comment:
This tries to copy the `TimeStep` class we have to the readers right? If yes then you can directly make a equivalent class for the auxillary reader.